server/src/main/java/org/apache/druid/server/QueryResultPusher.java

```diff
@@ -121,6 +128,11 @@ public Response push()
       final Response.ResponseBuilder startResponse = resultsWriter.start();
       if (startResponse != null) {
         startResponse.header(QueryResource.QUERY_ID_RESPONSE_HEADER, queryId);
+
+        if (includeTrailerHeader) {
+          startResponse.header(HttpHeader.TRAILER.toString(), QueryResource.ERROR_MESSAGE_TRAILER_HEADER);
```

Comment:
what is the purpose of having this "header?

Reply:
As per standard, we need to specify the header names which will be part of Trailer. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Trailer